pkg/infra/observer.go

```diff
@@ -11,56 +12,49 @@ import (
 type Observer struct {
 	d      peer.Deliver_DeliverFilteredClient
 	logger *log.Logger
-	signal chan error
+	//signal chan error
 }
 
-func CreateObserver(channel string, node Node, crypto *Crypto, logger *log.Logger) *Observer {
-	deliverer, err := CreateDeliverFilteredClient(node)
+func CreateObserver(channel string, node Node, crypto *Crypto, logger *log.Logger) (*Observer, error) {
+	deliverer, err := CreateDeliverFilteredClient(node, logger)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 
 	seek, err := CreateSignedDeliverNewestEnv(channel, crypto)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 
 	if err = deliverer.Send(seek); err != nil {
-		panic(err)
+		return nil, err
 	}
 
 	// drain first response
 	if _, err = deliverer.Recv(); err != nil {
-		panic(err)
+		return nil, err
 	}
 
-	return &Observer{d: deliverer, signal: make(chan error, 10), logger: logger}
+	return &Observer{d: deliverer, logger: logger}, nil
 }
 
-func (o *Observer) Start(N int, now time.Time) {
-	defer close(o.signal)
+func (o *Observer) Start(N int, now time.Time, ec chan error) {
+	defer close(ec)
```

Comment:
see previous comments about multi-sender caveat for such pattern

Reply:
will deal with previous one.